include_core/AtomicSupport.hpp

```diff
@@ -307,7 +311,9 @@ class VM_AtomicSupport
 		*/
 #elif defined(ARM) /* defined(J9HAMMER) */
 		__sync_synchronize();
-#else /* defined(ARM) */
+#elif defined(AARCH64) /* defined(ARM) */
+		__asm __volatile ("dmb ishld":::"memory")
+#else /* defined(AARCH64) */
```

Comment:
line  231, 270, and 315
I have used internal memory barriers, but is this restrictive enough ?
(these can also be used for ARM! )
there is three shareable domain, inner, outer or  all
see http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0802b/CIHGHHIE.html